producer.go

```diff
@@ -342,12 +342,21 @@ func (w *Producer) popTransaction(frameType int32, data []byte) {
 }
 
 func (w *Producer) transactionCleanup() {
-	// clean up transactions we can easily account for
-	for _, t := range w.transactions {
-		t.Error = ErrNotConnected
-		t.finish()
+l:
+	for {
+		select {
+		case data := <-w.responseChan:
+			w.popTransaction(FrameTypeResponse, data)
```

Comment:
Should these be popped and finished with `ErrNotConnected`, like below?

Reply:
I think the release code like below：


``` go

func (w *Producer) transactionCleanup() {
l:
	for {
		select {
		case data := <-w.responseChan:
			w.popTransaction(FrameTypeResponse, data)
		case data := <-w.errorChan:
			w.popTransaction(FrameTypeError, data)
		default:
			// clean up transactions we can easily account for
			for _, t := range w.transactions {
				t.Error = ErrNotConnected
				t.finish()
			}
			w.transactions = w.transactions[:0]
			break l
		}
	}
        ... ...
}
```

The  transaction poped in ```chan w.responseChan``` branch means successfully sent, so transaction's error reason is empty and should not be set.
The  transaction poped in ```chan w.errorChan``` branch means sent failed, should be send again, and transaction's error reason should be set to the real reason, so user can know what happened.